Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -316,46 +319,38 @@ void OpenALStream::SoundLoop()
 
       if (float32_capable)
       {
-        alBufferData(uiBuffers[nextBuffer], AL_FORMAT_51CHN32, dpl2,
+        alBufferData(uiBuffers[nextBuffer], AL_FORMAT_51CHN32, dpl2.data(),
                      nSamples * FRAME_SURROUND_FLOAT, ulFrequency);
       }
       else if (fixed32_capable)
       {
-        int surround_int32[OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS];
+        std::vector<int> surround_int32(OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS);
 
         for (u32 i = 0; i < nSamples * SURROUND_CHANNELS; ++i)
         {
           // For some reason the ffdshow's DPL2 decoder outputs samples bigger than 1.
           // Most are close to 2.5 and some go up to 8. Hard clamping here, we need to
           // fix the decoder or implement a limiter.
-          dpl2[i] = dpl2[i] * (INT64_C(1) << 31);
-          if (dpl2[i] > INT_MAX)
-            surround_int32[i] = INT_MAX;
-          else if (dpl2[i] < INT_MIN)
-            surround_int32[i] = INT_MIN;
-          else
-            surround_int32[i] = (int)dpl2[i];
+          dpl2[i] = dpl2[i] * (static_cast<s64>(1) << 31);
+		  surround_int32[i] = static_cast<s32>(MathUtil::Clamp(
```

Comment:
You could avoid a lot of these casts by doing `surround_int32[i] = MathUtil::Clamp(dpl2[i], -1.0, 1.0) * INT32_MAX` (the minimum would be `INT32_MIN - 1` instead of the current `INT32_MIN`).